src/SimpleDTO.php

```diff
@@ -350,6 +342,18 @@ protected function validatedData(): array
         return $result;
     }
 
+    /**
+     *  Check cast is callable or instance of Castable.
+     */
+    protected function getCast($cast, string $key, mixed $value): mixed
+    {
+        if($cast instanceof  \WendellAdriel\ValidatedDTO\Casting\Castable){
+            return $cast->cast($key, $value);
+        }
+
+        return  is_callable($cast) ? $cast($key, $value) : $value;
```

Comment:
Check if it's callable, and if it's not you should throw the `CastTargetException`

Reply:
@WendellAdriel Thank you ❤️ for your guideline. I am changing.

Reply:
@WendellAdriel , I have fixed all concerns from you.